api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -40,6 +55,36 @@
 	@Autowired
 	private ObservationCategoryMap categoryMap;
 	
+	@Autowired
+	@Getter(AccessLevel.PUBLIC)
+	@Setter(AccessLevel.PUBLIC)
+	@Qualifier("sessionFactory")
+	private SessionFactory sessionFactory;
```

Comment:
Do we really need this? I think we should be able to use the session factory instance from the BaseFhirDao class.

Reply:
Yes we'll do that, but we need to change the access modifier of `sessionFactory` in  BaseFhirDao

Reply:
It's already public

Reply:
FhirObservationDaoImpl was not able to access it until I changed `private` to `protected`.

Reply:
Shouldn't we use the getter instead of accessing the property directly?

Reply:
Yes, that seems correct.